cmake/EthCompilerSettings.cmake

```
@@ -197,6 +197,8 @@ if (SANITIZE)
 	# -fsanitize-address-use-after-scope throws an error when a variable is used beyond its scope
 	if (san STREQUAL "address")
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-omit-frame-pointer -fsanitize=address -fsanitize-address-use-after-scope")
+	elseif (san STREQUAL "undefined")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unsigned-integer-overflow,unreachable,vla-bound,vptr -fno-sanitize-recover=array-bounds,bool,builtin,enum,float-divide-by-zero,function,integer-divide-by-zero,null,object-size,return,returns-nonnull-attribute,shift,signed-integer-overflow,unreachable,vla-bound,vptr")
```

This line is very long and you also have two lists that seem identical at a glance but are not. I think this would be a much better way to format it:
```suggestion
		string(JOIN "," _sanitizers, ""
			array-bounds
			bool
			builtin
			enum
			float-divide-by-zero
			function
			integer-divide-by-zero
			null
			object-size
			return
			returns-nonnull-attribute
			shift
			signed-integer-overflow
			unreachable
			vla-bound
			vptr
		)
		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=${_sanitizers} -fno-sanitize-recover=${_sanitizers},unsigned-integer-overflow")
```
I did not test it in practice so it may need some tweaks (not sure if the variable in JOIN needs to be declared first for example) but something long these lines should work.